src/parcsr_ls/ads.c

```diff
@@ -5,22 +5,32 @@
  * SPDX-License-Identifier: (Apache-2.0 OR MIT)
  ******************************************************************************/
 
+#include "_hypre_onedpl.hpp"
 #include "_hypre_parcsr_ls.h"
 #include "float.h"
 #include "ams.h"
 #include "ads.h"
 #include "_hypre_utilities.hpp"
 
-#if defined(HYPRE_USING_CUDA) || defined(HYPRE_USING_HIP)
+#if defined(HYPRE_USING_GPU)
+#if defined(HYPRE_USING_SYCL)
+SYCL_EXTERNAL
```

Comment:
Do we need `SYCL_EXTERNAL` because the kernels are called from another file?

Reply:
Correct